src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -428,7 +438,14 @@ MultiTaskTrackerExecute(Job *job)
 	 * If we previously broke out of the execution loop due to a task failure or
 	 * user cancellation request, we can now safely emit an error message.
 	 */
-	if (taskFailed)
+	if (taskFailed && sizeLimitIsExceeded)
+	{
+		ereport(ERROR, (errmsg("failed to execute task %u", failedTaskId),
+						errhint("the max intermediate result size is exceeded "
```

Comment:
Maybe move this line into `errmsg`. Most clients / ORMs / etc. only expose the error message, so `failed to execute task` is not very helpful.

(yes, we do it for other errors, and that's bad)